### DISCUSSION TITLE
how to parse an optional number

### DISCUSSION THREAD

**Discussion by jdgamble555** [open-circle/valibot#430] [General]
How would I parse an optional number?

```ts
const postSchema = object({
  ...
  category_order: optional(number()),
  ...
});

const formData = decode_formdata(await request.formData(), {
    numbers: ['category_order']
});
```

This does not seem to work when optional, as I get this message:
```
category_order - NaN - Invalid type
```
when checking the error:
```ts
const msg = result.issues[0].path[0].key + ' - ' + result.issues[0].path[0].value + ' - ' + result.issues[0].message;
```

J

**Comment by fabian-hiller** [comment]
This is a problem with the `FormData` decoding library you are using. If you are using [my library](https://github.com/fabian-hiller/decode-formdata), please file an issue there. Without checking the source of `decode-formdata`, I think we currently do not support multiple data types on a single file.

**Reply by jdgamble555** [reply]
Actually that is your library, I had to rename it because I had another decode function. Will file issue.
